lexer/lexer.go

```diff
@@ -77,13 +77,20 @@ func (l *Lexer) NextToken() *token.Token {
 		}
 		return token.ConstantTokenChar(ch)
 	case '+', '-':
-		if nextChar == ch {
+		if nextChar == ch || nextChar == '=' {
 			l.pos++
 			return token.ConstantTokenChar2(ch, nextChar) // increment/decrement
 		}
 		return token.ConstantTokenChar(ch)
-	case '%', '*', ';', ',', '{', '}', '(', ')', '[', ']', '^', '~':
+	case '%', ';', ',', '{', '}', '(', ')', '[', ']', '^', '~':
```

Comment:
%= ^= ~= should also work I think - this being said I am curious why is * here and not /

Reply:
ic below the / has to handle comments so that's why it was special already

we can add the %= etc later (including &= |= )